pkg/sql/instrumentation.go

```diff
@@ -755,6 +755,9 @@ func (ih *instrumentationHelper) Finish(
 		// of the transaction if the transaction matches the request.
 		// NB: It is safe for bundle to be empty here.
 		txnHelper.AddStatementBundle(ctx, ast, uint64(ih.fingerprintId), ih.fingerprint, bundle)
+		if ih.implicitTxn {
+			txnHelper.diagnosticsCollector.UpdateState(txnDiagnosticsReadyToFinalize)
+		}
```

Comment:
is there no hook for implicit transaction to commit that we could tie into? just wondering about the placement of this update.

Reply:
@dhartunian @jasonlmfong  implicit transactions are auto committed before the bundle is created or added here, so I don't think we should update the state there. Instead of doing this state update here, i made a change to update the state to`readyToFinalize` when finalizing the collection if we are still in the `inProgress` state. See the commit message for more details